src/pages/Admin/hooks.ts

```diff
@@ -0,0 +1,83 @@
+import { useMemo } from "react";
+import { useQueryWithClient } from "@deskpro/app-sdk";
+import { getCreateMeta, getFields } from "../../api/api";
+import type { DropdownItemType } from "@deskpro/deskpro-ui";
+import type { ProjectElement, Issuetype } from "../../api/types/createMeta";
+import type { FieldMeta } from "../../api/types/types";
+import type { Settings } from "../../types";
+
+type UseMetadata = (settings?: Settings, projectId?: ProjectElement["id"]) => {
+  isLoading: boolean;
+  projects: ProjectElement[];
+  projectOptions: Array<DropdownItemType<ProjectElement["id"]>>;
+  issueTypes: Issuetype[];
+  issueTypeOptions: Array<DropdownItemType<Issuetype["id"]>>;
+  fields: FieldMeta[];
+};
+
+const useMetadata: UseMetadata = (settings, projectId) => {
+  const isValidSettings = Boolean(settings?.domain) && Boolean(settings?.username) && Boolean(settings?.api_key);
+
+  const fields = useQueryWithClient(
+    ["fields"],
+    (client) => getFields(client, settings),
+    { enabled: isValidSettings },
+  );
+
+  const createMeta = useQueryWithClient(
+    ["createMetaQuery"],
+    (client) => getCreateMeta(client, settings),
+    { enabled: isValidSettings },
+  );
+
+  const projectOptions: Array<DropdownItemType<ProjectElement["id"]>> = useMemo(() => {
+    const projects = createMeta.data?.projects ?? [];
+
+    if (!projects.length) {
+      return [{ type: "header", label: "No project(s) found" }];
+    }
+
+    return projects.map((p) => ({
```

Comment:
Rename `p` to something more descriptive, like `project`

Reply:
fixed